.ci/dockerfiles/Dockerfile.rocm

```
@@ -0,0 +1,608 @@
+    cp /tmp/gtest-parallel/* "/out${NIXL_INSTALL_DIR}/bin/"
+
+# ============================================================================
+# ucx-build  (independent, parallel with everything)
+# MUTABLE ref v1.21.x -> layer cache may go stale; audit on refresh.
+# Installs to ${UCX_INSTALL_DIR} (defaults to ${NIXL_INSTALL_DIR}) -> /out/opt/nixl.
+# ROCm path: no CUDA/gdrcopy/UCCL.
+# ============================================================================
+FROM ${BUILDER_BASE_IMAGE} AS ucx-build
+ARG NIXL_INSTALL_DIR=/opt/nixl
+ARG UCX_VERSION=v1.21.x
+RUN set -eux; \
```

Hadolint (2.14.0)

[warning] Use WORKDIR to switch to a directory

(DL3003)